### PR TITLE
NuGetAuditMode default change respects SdkAnalysisLevel

### PR DESCRIPTION
**Pull request by zivkan**

### CODE REVIEW

File: docs/reference/errors-and-warnings/NU1901-NU1904.md [hunk 1]
```diff
@@ -51,3 +51,7 @@ For example `<NuGetAuditSuppress Include="https://github.com/advisories/GHSA-g3q
 
 If you do not want NuGet to check for packages with known vulnerabilities during restore, add `<NuGetAudit>false</NuGetAudit>` inside a `<PropertyGroup>` in your project file, or a [`Directory.Build.props` file](/visualstudio/msbuild/customize-by-directory).
 If you would like to run NuGet Audit on developer machines, but disable it on CI pipelines, you can take advantage of MSBuild importing environment variables, and create a NuGetAudit environment variable set to `false` in your pipeline definition.
+
+In NuGet 6.12 (Visual Studio/MSBuild 17.12 and .NET 9.0.100 SDK), NuGet changed the default for `NuGetAuditMode` to `all`, which means report on transitive packages with known vulnerabilities.
+The value can be explicitly set to `direct` to revert back to .NET 8's default.
+Alternatively, [the property `SdkAnalysisLevel`](/dotnet/core/project-sdk/msbuild-props#sdkanalysislevel) can be set to `8.0.400` to lock multiple NuGet and .NET SDK features to .NET 8.0.400 defaults.
```
Comment (nkolev92) [inline docs/reference/errors-and-warnings/NU1901-NU1904.md:57]: Should we call out the scenario which is, if you prefer not to address it now, you can temporarily set etc. 

The spirit of SDK Analysis level is until, behave like an older SDK until I have time to address it.

File: docs/concepts/Auditing-Packages.md [hunk 2]
```diff
@@ -47,6 +46,7 @@ We recommend that audit is configured at a repository level.
 | NuGetAuditMode | all (1) | `direct` and `all` | If you'd like to audit both top-level and transitive dependencies, you can set the value to `all`. NuGetAuditMode is not applicable for packages.config projects |
 | NuGetAuditLevel | low | `low`, `moderate`, `high`, and `critical` | The minimum severity level to report. If you'd like to see `moderate`, `high`, and `critical` advisories (exclude `low`), set the value to `moderate` |
 | NuGetAudit | true | `true` and `false` | If you wish to not receive security audit reports, you can opt-out of the experience entirely by setting the value to `false` |
+| [SdkAnalysisLevel](/dotnet/core/project-sdk/msbuild-props#sdkanalysislevel) | The version of the .NET SDK that is installed | `8.0.400`, `9.0.100` | Introduced in .NET 9, this property can be used to pin multiple NuGet and .NET SDK features to their defaults at the specified SDK version. |
```
Comment (nkolev92) [inline docs/concepts/Auditing-Packages.md:49]: Same disclaimer about intentions of this being temporary.

**Review by JonDouglas — APPROVED** [review]
;lgtm ship!

File: docs/concepts/Auditing-Packages.md [hunk 3]
```diff
@@ -44,11 +43,11 @@ We recommend that audit is configured at a repository level.
...
+| NuGetAuditMode | all | `direct` and `all` | If you'd like to audit top-level dependencies only, you can set the value to `direct`. NuGetAuditMode is not applicable for packages.config projects.  |
 | NuGetAuditLevel | low | `low`, `moderate`, `high`, and `critical` | The minimum severity level to report. If you'd like to see `moderate`, `high`, and `critical` advisories (exclude `low`), set the value to `moderate` |
 | NuGetAudit | true | `true` and `false` | If you wish to not receive security audit reports, you can opt-out of the experience entirely by setting the value to `false` |
 
-(1) NuGetAuditMode defaulted to `direct` when it was introduced in [the .NET 8.0.100 SDK and VS 17.8](../release-notes/NuGet-6.8.md). In [.NET 9.0.100 SDK and VS 17.12](../release-notes/NuGet-6.12.md) the default changed to `all`.
+Note: In .NET 8, the default value of NuGetAuditMode is `direct`. Therefore, setting [SdkAnalysisLevel](/dotnet/core/project-sdk/msbuild-props#sdkanalysislevel) to `8.0.400` changes the default value of NuGetAuditMode accordingly.
```
Comment (zivkan) [inline docs/concepts/Auditing-Packages.md:50]:
```suggestion
Note: In .NET 8, the default value of NuGetAuditMode is `direct`.
Therefore, setting [SdkAnalysisLevel](/dotnet/core/project-sdk/msbuild-props#sdkanalysislevel) to `8.0.400` changes the default value of NuGetAuditMode accordingly.
```

One sentence per line makes smaller diffs in the future.  Github also doesn't highlight changes when lines are too long, so one sentence per line helps with that too.

**Review by nkolev92 — APPROVED** [review]

**Pull request merged**